network/src/protocols/rpc/mod.rs

```diff
@@ -191,9 +192,9 @@ impl InboundRpcs {
     }
 
     /// Handle a new inbound `RpcRequest` message off the wire.
-    pub async fn handle_inbound_request(
+    pub fn handle_inbound_request(
         &mut self,
-        peer_notifs_tx: &mut channel::Sender<PeerNotification>,
+        peer_notifs_tx: &mut diem_channel::Sender<ProtocolId, PeerNotification>,
```

Comment:
this is definitely not super clear... `channel` is a single async FIFO queue that provides backpressure, i.e., `tx.send(msg).await` will wait until there is capacity before finally enqueuing. In contrast, `diem_channel` is a set of non-async queues (one per `ProtocolId` in this case). these queues _don't_ provide backpressure, i.e., `tx.send(msg)` (note the lack of `.await`) will just drop a message (the current msg if it's in FIFO mode) and immediately return.

of course, the previous state wasn't providing any backpressure either, since there was a no-backpressure `diem_channel` feeding into a backpressured `channel`

```
              diem_channel                          channel 
PeerManager <--------------> NetworkProvider task <---------> Peer task
```

much of this diff is removing the intermediate `channel`s

```
              diem_channel
PeerManager <--------------> Peer task
```

Reply:
Okay, that makes more sense to me now.  We'll have to visit later if we want backpressure again or not, but since this didn't do that really anyways it didn't make a difference :P